cloudamqp/resource_cloudamqp_vpc_connect.go

```diff
@@ -0,0 +1,245 @@
+package cloudamqp
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/84codes/go-api/api"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var (
+	AWS_ARN_VALIDATE_RE, _        = regexp.Compile(`\Aarn:aws:iam::\d{12}:(root|((user|role)(/[^/]+)?))\z`)
+	AZURE_SUBS_VALIDATE_RE, _     = regexp.Compile(`\A[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\z`)
+	GCP_PROJECT_ID_VALIDATE_RE, _ = regexp.Compile(`\A[a-z][0-9a-z-]{4,28}[0-9a-z]\z`)
+)
+
+func resourceVpcConnect() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVpcConnectCreate,
+		Read:   resourceVpcConnectRead,
+		Update: resourceVpcConnectUpdate,
+		Delete: resourceVpcConnectDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The CloudAMQP instance identifier",
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The region where the CloudAMQP instance is hosted",
+			},
+			"allowed_principals": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "List of allowed prinicpals used by AWS",
+			},
+			"approved_subscriptions": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "List of approved subscriptions used by Azure",
+			},
+			"allowed_projects": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "List of allowed projects used by GCP",
+			},
+			"active_zones": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed:    true,
+				Description: "Covering availability zones used when creating an endpoint from other VPC. [AWS]",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the Private Service Connect [enabled, pending, disabled]",
+			},
+			"service_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Service name (alias for Azure) of the PrivateLink.",
+			},
+			"sleep": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     10,
```

Comment:
I think we can handling polling more frequently, to give quicker feedback

```suggestion
				Default:     2,
```

Reply:
Do we really want more frequent requests? Previous default has been set to 30 s. Think at least we want to go down to 10 s.

Reply:
Yes, there's no harm doing a request every 2 seconds IMHO, but if we do know that it will never complete in less than X seconds, we don't need to poll more often than that. I'll state that I don't know the value of X :)

We can go with 5 seconds? :)

Reply:
Still thinks 10 sec in this case is good. During testing in one-go workflow, we still need to wait on post bootstrap (locally for AWS takes about 150 sec) before the PrivateLink/Private Service Connect request is started to get handeld by our backend. Then handle the request takes about 1-3 minutes.